File: core/src/main/clojure/com/mstream/rules_engine/java/evaluation_result.clj
(ns com.mstream.rules-engine.java.evaluation-result)


(defrecord EvaluationResult
  [finished?
   gatheredFacts
   matchedRule
   requestedFacts]
  com.mstream.rulesengine.core.EvaluationResult
  (isFinished [_]
    finished?)
  (getGatheredFacts [_]
    gatheredFacts)
  (getMatchedRule [_]
    matchedRule)
  (getRequestedFacts [_]
    requestedFacts)
  (toString [_]
    (str {:finished?      finished?
          :gatheredFacts  gatheredFacts
          :matchedRule    matchedRule
          :requestedFacts requestedFacts})))

File: core/src/main/java/com/mstream/rulesengine/core/EvaluationResult.java
package com.mstream.rulesengine.core;


import java.util.Map;
import java.util.Set;


public interface EvaluationResult {

    boolean isFinished();

    Rule getMatchedRule();

    Map<String, Boolean> getGatheredFacts();

    Set<String> getRequestedFacts();
}
